Rename `v1` to `v1alpha7`, and use `v1` from v0.6.x

tiyash-basu-frequenz opened the pull request:
closes #349

tiyash-basu-frequenz committed:
Rename `v1` package to `v1alpha7`

This commit makes the following changes:
- Renames the `v1` package to `v1alpha7` in the proto files.
- Renames the `v1` directory to `v1alpha7` in the proto files.
- Renames the `v1` directory to `v1alpha7` in the `py` directory.
- Rename the test file `test_common.py` to `test_common_v1alpha7.py`.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

tiyash-basu-frequenz committed:
Re-create the v1 package with the contents of the v0.6.x branch

This commit re-creates the `v1` package with the contents of the
`v0.6.x` branch, which is the latest stable version of the API. The
previous `v1` package has been renamed to `v1alpha7`.

This means that any client or server code that was using the v0.6.x
branch can now use the main branch without any changes, as the
`v1` package now contains the same contents as the `v0.6.x` branch.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

tiyash-basu-frequenz committed:
Fix a few typos in documentation

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

tiyash-basu-frequenz commented:
Waiting on https://github.com/frequenz-floss/frequenz-repo-config-python/pull/418 getting merged and incorporated into this repo.

llucax commented:
I created https://github.com/frequenz-floss/frequenz-api-common/pull/375 to not mix both issues, as this PR is already *disruptive* enough :scream:

llucax committed:
Add missing `__init__.py` files

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax committed:
Fix copyright and docstring for `__init__.py` files

Some files where missing the copyright notice completely, some files
have the wrong copyright year, some files had some TODOs in the
docstring and most had a bad or outdated docstring.

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

florian-wagner-frequenz commented:
Guess I'll approve to get this moving

tiyash-basu-frequenz commented:
Thanks @florian-wagner-frequenz :)
